lab_1_classify_by_unigrams/main.py

```diff
@@ -19,6 +26,22 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    token_frequencies = {}
+    total_tokens = len(tokens)
+    if all(isinstance(token, str) for token in tokens):
```

Comment:
Consider doing an early return here: if the requirement is unmet, immediately escape the function